File: packages/api/aqapi/shared/entity/sort_order.py
from dataclasses import dataclass
from typing import override

from aqapi.core.domain.value_object.base_value_object import BaseValueObject
from aqapi.core.messages.locale_string import LocaleString


@dataclass
class SortOrder(BaseValueObject):
    """ソート順序の値オブジェクト"""
    
    _value: int

    def __init__(self, value: int):
        super().__init__(value)

    def _validate(self) -> None:
        self._validator.required()
        self._validator.integer()

    @property
    @override
    def _value_name(self) -> LocaleString:
        return LocaleString(
            ja="ソート順序",
            en="Sort Order"
        )

File: packages/flutter/lib/shared/domain/value_object/sort_order.dart
import 'package:allowance_questboard/core/messages/locale_string.dart';
import 'package:allowance_questboard/core/value_object/base_state_object.dart' show BaseValueObject;

/// ソート順の値オブジェクト
class SortOrder extends BaseValueObject<int> {
  SortOrder(super.value);
  
  @override
  void validate() {
    validator.required();
    validator.integer();
    validator.minValue(0);
  }
  
  @override
  LocaleString get valueName => const LocaleString(
    ja: "ソート順",
    en: "Sort Order",
  );
}
